addons/website/models/ir_http.py

```diff
@@ -163,7 +163,7 @@ def _get_public_users(cls):
         public_users = super()._get_public_users()
         website = request.env(user=SUPERUSER_ID)['website'].with_context(lang='en_US').get_current_website()  # sudo
         if website:
-            public_users.append(website._get_cached('user_id'))
+            public_users.append(website.user_id.id)
```

Comment:
This cleanup is really nice.  It makes the optimization completely transparent :+1: